planner/core/find_best_task.go

```diff
@@ -804,6 +807,13 @@ func (ds *DataSource) convertToTableScan(prop *property.PhysicalProperty, candid
 	if prop.ExpectedCnt < ds.stats.RowCount {
 		count, ok, corr := ds.crossEstimateRowCount(path, prop.ExpectedCnt, candidate.isMatchProp && prop.Items[0].Desc)
 		if ok {
+			// TODO: actually, before using this count as the estimated row count of table scan, we need additionally
+			// check if count < row_count(first_region | last_region), and use the larger one since we build one copTask
```

Comment:
Do we really need this todo? Coprocessor also breaks when it scans the first needed row?

Reply:
> Coprocessor also breaks when it scans the first needed row?

Hmm I don't understand this comment, could you please elaborate more on this? I am not saying that we modify coprocessor to return just needed rows, I mean that we should set rowCount as:
```golang
rowCount = math.Max(count, regionRowCount)
// regionRowCount is the row count of first region or last region based on ASC/DESC
```